api/v1beta1/glance_types.go

```diff
@@ -0,0 +1,182 @@
+	GlanceAPIExternal GlanceAPISpec `json:"glanceAPIExternal"`
+}
+
+// PasswordSelector to identify the DB and AdminUser password from the Secret
+type PasswordSelector struct {
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="GlanceDatabasePassword"
+	// Database - Selector to get the glance database user password from the Secret
+	// TODO: not used, need change in mariadb-operator
+	Database string `json:"database,omitempty"`
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="GlancePassword"
+	// Database - Selector to get the glance service password from the Secret
+	Service string `json:"admin,omitempty"`
+}
+
+// GlanceDebug defines the observed state of GlanceAPIDebug
+type GlanceDebug struct {
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=false
+	// DBSync enable debug
+	DBSync bool `json:"dbSync,omitempty"`
+}
+
+// GlanceStatus defines the observed state of Glance
+type GlanceStatus struct {
+	// Map of hashes to track e.g. job status
+	Hash map[string]string `json:"hash,omitempty"`
+
+	// API endpoint
+	APIEndpoints map[string]string `json:"apiEndpoint,omitempty"`
+
+	// ServiceID
+	ServiceID string `json:"serviceID,omitempty"`
+
+	// Conditions
+	Conditions condition.Conditions `json:"conditions,omitempty" optional:"true"`
+
+	// Glance Database Hostname
+	DatabaseHostname string `json:"databaseHostname,omitempty"`
+
+	// ReadyCount of internal and admin Glance API instance
+	GlanceAPIInternalReadyCount int32 `json:"cinderAPIReadyInternalCount,omitempty"`
+
+	// ReadyCount of external and admin Glance API instance
+	GlanceAPIExternalReadyCount int32 `json:"cinderAPIReadyExternalCount,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:printcolumn:name="Status",type="string",JSONPath=".status.conditions[0].status",description="Status"
+//+kubebuilder:printcolumn:name="Message",type="string",JSONPath=".status.conditions[0].message",description="Message"
+
+// Glance is the Schema for the glances API
+type Glance struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   GlanceSpec   `json:"spec,omitempty"`
+	Status GlanceStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// GlanceList contains a list of Glance
+type GlanceList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Glance `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&Glance{}, &GlanceList{})
+}
+
+// IsReady - returns true if service is ready to serve requests
+func (instance Glance) IsReady() bool {
+	// Ready when:
+	// - Service is registered in Keystone
+	// - Both the internal and external API endpoints are ready (count >= 1)
+	return instance.Status.ServiceID != "" && instance.Status.GlanceAPIInternalReadyCount >= 1 && instance.Status.GlanceAPIExternalReadyCount >= 1
```

Comment:
Fair point, I will prune it

Reply:
right, this seems in line with the check we do in the cinder operator, but since here we only have Internal and External checking `instance.Status.GlanceAPIInternalReadyCount >= 1` and  `instance.Status.GlanceAPIExternalReadyCount >= 1` is fair enough

Reply:
Removed `instance.Status.ServiceID != ""`